new-log-viewer/src/services/decoders/JsonlDecoder/utils.ts

```diff
@@ -0,0 +1,92 @@
+import dayjs, {Dayjs} from "dayjs";
+import utc from "dayjs/plugin/utc";
+
+import {
+    JsonObject,
+    JsonValue,
+} from "../../../typings/js";
+import {
+    INVALID_TIMESTAMP_VALUE,
+    LOG_LEVEL,
+} from "../../../typings/logs";
+
+
+// eslint-disable-next-line import/no-named-as-default-member
+dayjs.extend(utc);
+
+/**
+ * Determines whether the given value is a `JsonObject` and applies a TypeScript narrowing
+ * conversion if so.
+ *
+ * Reference: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
+ *
+ * @param value
+ * @return A TypeScript type predicate indicating whether `value` is a `JsonObject`.
+ */
+const isJsonObject = (value: JsonValue): value is JsonObject => {
+    return "object" === typeof value && null !== value;
+};
+
+/**
+ * Converts a field into a log level if possible.
+ *
+ * @param field
+ * @return The log level or `LOG_LEVEL.NONE` if the field couldn't be converted.
+ */
+const convertToLogLevelValue = (field: JsonValue | undefined): LOG_LEVEL => {
+    let logLevelValue = LOG_LEVEL.NONE;
+
+    if ("undefined" === typeof field) {
+        return logLevelValue;
+    }
+
+    const logLevelName = "object" === typeof field ?
+        JSON.stringify(field) :
```

Comment:
Mind explaining why we want to convert an object into JSON then compare it against a list of log levels?

Reply:
i.e. can we type-check against "string" and if not return LOG_LEVEL.NONE as done in L40?

Reply:
So sorry - I wrote this code but I completely forgot why it was written that way till just now. 

The reason of converting the logLevelField into a json string when it's an object is because the log level field could look like this:
```json
{
   "name": "INFO",
   "value": 20
}
```
rather than a regular string "INFO" for example. 

With the json string conversion, we will be able to extract the log level from such objects.

Let's document this inline.